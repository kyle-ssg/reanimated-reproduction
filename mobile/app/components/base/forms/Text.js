import propTypes from 'prop-types';
import ReactNative from 'react-native';
import React, { Component } from 'react';

export default class Text extends Component {
  static displayName = 'Text';

  static propTypes = {
      style: propTypes.any,
      children: propTypes.node,
  };

  render() {
      const {
          props: { style, children },
      } = this;
      return (
          <ReactNative.Text {...this.props} style={[Styles.text, style]}>
              {children}
          </ReactNative.Text>
      );
  }
}
