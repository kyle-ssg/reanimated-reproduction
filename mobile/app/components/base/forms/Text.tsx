import ReactNative from 'react-native';
import React, { FunctionComponent } from 'react'; // we need this to make JSX compile

type ComponentType = ReactNative.TextProps & {}

const TheComponent: FunctionComponent<ComponentType> = (props) => {
  return (
      <ReactNative.Text {...props} style={[Styles.text, props.style]}>
          {props.children}
      </ReactNative.Text>
  );
};

export default TheComponent;
