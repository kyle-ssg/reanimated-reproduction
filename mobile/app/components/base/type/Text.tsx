import { Text, TextProps } from 'react-native'
import { FC } from 'react'
import { asStyleProp, cn } from '../../../style/_style_screen'
import Animated from 'react-native-reanimated'

const textSizeClass = asStyleProp({
  regular: '',
  medium: 'textMedium',
  small: 'textSmall',
  h1: 'h1',
  h2: 'h2',
  h3: 'h3',
  h4: 'h4',
})

const textWeightClass = asStyleProp({
  regular: '',
  bold: 'textBold',
})

export type TextType = TextProps & {
  muted?: boolean
  light?: boolean
  size?: keyof typeof textSizeClass
  weight?: keyof typeof textWeightClass
  animated?: boolean
}
const TextComponent: FC<TextType> = (props) => {
  // @ts-ignore
  const TextComponent: typeof Animated.Text = props.animated
    ? Animated.Text
    : Text
  return (
    <TextComponent
      {...props}
      style={cn(
        {
          text: true,
          textMuted: props.muted,
          textWhite: props.light,
        },
        Styles[textSizeClass[props.size]],
        Styles[textWeightClass[props.weight]],
        props.style,
      )}
    >
      {props.children}
    </TextComponent>
  )
}

export default TextComponent
