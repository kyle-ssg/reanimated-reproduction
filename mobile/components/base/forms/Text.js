'use strict';

var Button = Component({
  displayName: 'Text',
  render: function () {
    return <ReactNative.Text style={[Styles.text, this.props.style]}>{this.props.children}</ReactNative.Text>
  },

});


module.exports = Button;
