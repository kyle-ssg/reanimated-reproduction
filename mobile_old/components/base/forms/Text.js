var Text = Component({
  displayName: 'Text',
  render: function () {
    return <ReactNative.Text style={[Styles.text, this.props.style]}>{this.props.children}</ReactNative.Text>;
  }
});

Text.propTypes = {
  style: React.PropTypes.any,
  children: OptionalNode,
};

module.exports = Text;
