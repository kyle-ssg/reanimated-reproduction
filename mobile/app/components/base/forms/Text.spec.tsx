import * as React from 'react'
import Text from './Text'
import renderer from 'react-test-renderer'

describe('<Text />', () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<Text />)
    expect(tree).toMatchSnapshot()
  })
})
