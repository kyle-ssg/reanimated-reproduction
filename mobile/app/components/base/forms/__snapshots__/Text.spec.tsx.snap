// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "#333",
        "fontFamily": "System",
        "fontSize": 16,
      },
      undefined,
    ]
  }
/>
`;
